File: src/main/java/no/nav/fo/veilarbregistrering/bruker/pdl/hentPerson/PdlGeografiskTilknytning.java
package no.nav.fo.veilarbregistrering.bruker.pdl.hentPerson;

public class PdlGeografiskTilknytning {

    private GtType gtType;
    private String gtKommune;
    private String gtBydel;
    private String gtLand;

    public GtType getGtType() {
        return gtType;
    }

    public void setGtType(GtType gtType) {
        this.gtType = gtType;
    }

    public String getGtKommune() {
        return gtKommune;
    }

    public void setGtKommune(String gtKommune) {
        this.gtKommune = gtKommune;
    }

    public String getGtBydel() {
        return gtBydel;
    }

    public void setGtBydel(String gtBydel) {
        this.gtBydel = gtBydel;
    }

    public String getGtLand() {
        return gtLand;
    }

    public void setGtLand(String gtLand) {
        this.gtLand = gtLand;
    }

    @Override
    public String toString() {
        return "PdlGeografiskTilknytning{" +
                "gtType=" + gtType +
                ", gtKommune='" + gtKommune + '\'' +
                ", gtBydel='" + gtBydel + '\'' +
                ", gtLand='" + gtLand + '\'' +
                '}';
    }
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/bruker/pdl/endepunkt/PdlGeografiskTilknytning.kt
package no.nav.fo.veilarbregistrering.bruker.pdl.endepunkt

enum class PdlGtType {
    KOMMUNE,
    BYDEL,
    UTLAND,
    UDEFINERT;
}

data class PdlGeografiskTilknytning(
    val gtType: PdlGtType,
    val gtKommune: String?,
    val gtBydel: String?,
    val gtLand: String?
)

data class PdlHentGeografiskTilknytningResponse(val data: PdlHentGeografiskTilknytning)

data class PdlHentGeografiskTilknytningRequest(val query: String, val variables: HentGeografiskTilknytningVariables)

data class HentGeografiskTilknytningVariables(val ident: String)

data class PdlHentGeografiskTilknytning(val hentGeografiskTilknytning: PdlGeografiskTilknytning)
